arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/raise/RaiseAccumulate.kt

```diff
@@ -1,4 +1,4 @@
-@file:OptIn(ExperimentalTypeInference::class, ExperimentalContracts::class)
```

Comment:
No more contracts?

Reply:
I got confused how we should use them, because they resulted in a couple of false-negatives of "Unreachable code" 😕

In this case of `catch` we should be able to tell the compiler that the second lambda "recovers" from the `Nothing` of the first lambda. It's a bit strange since the _inferred_ returned type is `Int` of the `catch` lambda. So I guess `callsInPlace` assumes _not-wrapped in try_?

<img width="535" alt="Screenshot 2023-03-06 at 18 49 39" src="https://user-images.githubusercontent.com/12424668/223190506-55e14dd1-f9db-4e37-a04a-bf93d8179217.png">

The second one is more complex.. since `either` is also `callsInPlace` and the last statement is an exception, but it's not throwing but _forever suspending_... And it exists early due to `raise` 😅 

<img width="700" alt="Screenshot 2023-03-06 at 18 43 59" src="https://user-images.githubusercontent.com/12424668/223189379-a6e0529e-d997-4bf6-9a74-6bd18e384fb7.png">

Not sure how we should deal with this best. WDYT @serras?

Reply:
I also noticed that `ResultKt` only specifies `AT_MOST_ONCE` and never `EXACTLY_ONCE`.

Reply:
Thanks for the thorough investigation, I never noticed that before. I assumed that this would help the compiler inline or optimize the code better, but it seems that it's reaching wrong conclusions...

Reply:
I am going to merge this PR, but I am continuing my search for a clear answer on this in Kotlin Slack. Re-asked my question in #library-development, https://kotlinlang.slack.com/archives/C8C4JTXR7/p1678213597746259.

It seems that `AT_MOST_ONCE` doesn't suffer from the _false-negative_ so I am going to re-add that back before merging.